src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -516,16 +516,29 @@ LedgerMaster::tryFill (
 /** Request a fetch pack to get to the specified ledger
 */
 void
-LedgerMaster::getFetchPack (LedgerIndex missingIndex,
+LedgerMaster::getFetchPack (LedgerIndex missing,
     InboundLedger::Reason reason)
 {
-    auto haveHash = getLedgerHashForHistory(
-        missingIndex + 1, reason);
+    auto haveHash {getLedgerHashForHistory(missing + 1, reason)};
     if (!haveHash || haveHash->isZero())
     {
-        JLOG (m_journal.error()) <<
-            "No hash for fetch pack. Missing Index " <<
-            std::to_string(missingIndex);
+        if (reason == InboundLedger::Reason::SHARD)
+        {
+            auto const shardStore {app_.getShardStore()};
+            auto const shardIndex {shardStore->seqToShardIndex(missing)};
+            if (missing < shardStore->lastLedgerSeq(shardIndex))
```

Comment:
does `shardStore->LastLedgerSeq()` return the last ledger index within the shard, or the ledger index 1 greater than that? If it really is the last index within the shard, then this should probably be a `<=`, right?